lib/src/view/tournament/tournament_list_screen.dart

```diff
@@ -0,0 +1,271 @@
+import 'package:collection/collection.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:intl/intl.dart';
+import 'package:lichess_mobile/src/model/common/chess.dart';
+import 'package:lichess_mobile/src/model/tournament/tournament.dart';
+import 'package:lichess_mobile/src/model/tournament/tournament_providers.dart';
+import 'package:lichess_mobile/src/styles/lichess_colors.dart';
+import 'package:lichess_mobile/src/styles/styles.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/utils/navigation.dart';
+import 'package:lichess_mobile/src/widgets/list.dart';
+import 'package:lichess_mobile/src/widgets/platform.dart';
+import 'package:lichess_mobile/src/widgets/platform_scaffold.dart';
+
+class TournamentListScreen extends ConsumerWidget {
+  const TournamentListScreen({super.key});
+
+  static Route<void> buildRoute(BuildContext context) {
+    return buildScreenRoute(
+      context,
+      title: context.l10n.tournaments,
+      screen: const TournamentListScreen(),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return switch (ref.watch(tournamentsProvider)) {
+      AsyncData(:final value) => PlatformWidget(
+        androidBuilder: (_) => _AndroidBody(tournaments: value),
+        iosBuilder: (_) => _CupertinoBody(tournaments: value),
+      ),
+      AsyncError(:final error) => Center(child: Text('Could not load tournaments: $error')),
+      _ => PlatformScaffold(
+        appBarTitle: Text(context.l10n.tournaments),
+        body: const Center(child: CircularProgressIndicator()),
+      ),
+    };
+  }
+}
+
+class _AndroidBody extends StatefulWidget {
+  const _AndroidBody({required this.tournaments});
+
+  final TournamentLists tournaments;
+
+  @override
+  State<_AndroidBody> createState() => _AndroidBodyState();
+}
+
+enum _ViewMode {
+  completed,
+  ongoing,
+  upcoming;
+
+  String l10n(BuildContext context) {
+    switch (this) {
+      case _ViewMode.completed:
+        // TODO l10n
+        return 'Completed';
+      case _ViewMode.ongoing:
+        return context.l10n.broadcastOngoing;
+      case _ViewMode.upcoming:
+        return context.l10n.broadcastUpcoming;
+    }
+  }
+}
+
+class _AndroidBodyState extends State<_AndroidBody> with TickerProviderStateMixin {
+  late final TabController _tabController;
+
+  @override
+  void initState() {
+    super.initState();
+    _tabController = TabController(initialIndex: 1, length: 3, vsync: this);
+  }
+
+  @override
+  void dispose() {
+    _tabController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```

Comment:
Nested `Scaffolds` are better to be avoided.

Reply:
How is it nested? the parent widget is a `PlatformScaffold` only while loading, but not when it has finished loading (and is displaying the body)

Reply:
Ah, I guess you mean there should only be one scaffold at the top, not in multiple places? Ok I also saw you other comment now, will do that